packages/subgraph/schema.graphql

```diff
@@ -1,181 +1,460 @@
   id: ID!
+
+  " The [ERC721 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-721) instance. "
   token: ERC721Contract!
+
+  " The DAO that holds tokens from a [ERC721 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-721). "
   dao: Dao!
+
+  " The NFTs from one ERC721 Contract held by the DAO. "
   tokenIds: [BigInt!]!
+
+  " The block timestamp of the last update (transfer) for this token balance. "
   lastUpdated: BigInt!
 }
 
+"""
+The ERC1155 Token balances of the DAO. Including all Token IDs and it's amounts held by the DAO per [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155).
+"""
 type ERC1155Balance implements TokenBalance @entity {
+  " The concatenation of DAO Smart Contract address and [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155) address. Joined by '_'. "
   id: ID!
+
+  " The [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155) instance. "
   token: ERC1155Contract!
+
+  " The DAO that holds tokens from a [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155). "
   dao: Dao!
+
+  " The amount and related information of the DAO held tokens per [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155). "
   balances: [ERC1155TokenIdBalance!]! @derivedFrom(field: "balance")
+
+  " The URI of the [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155). "
   metadataUri: String!
+
+  " The block timestamp of the last update (transfer) for this token balance. "
   lastUpdated: BigInt!
 }
 
+"""
+The DAO token balance for each token within an [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155)
+"""
 type ERC1155TokenIdBalance @entity {
+  # daoAddress_tokenAddress_tokenID"
+  " The concatenation of DAO Smart Contract address, [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155) and token ID. Joined by '_'. "
   id: ID!
+
+  #daoAddress+tokenAddress
+  " One to Many relationship linking with ERC1155Balance entity. "
   balance: ERC1155Balance!
+
+  " The ID for each different token in an [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155) held by the DAO. "
   tokenId: BigInt!
+
+  " The number of tokens held by the DAO per [ERC1155 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-1155) and token ID. "
   amount: BigInt!
+
+  " The block timestamp of the last update (transfer) for this token balance. "
   lastUpdated: BigInt!
 }
 
-# Token Transfers
+###########################
+##### Token Transfers #####
+###########################
 
+"""
+Classify the types of transfers from the DAO's point of view.
+"""
 enum TransferType {
-  ExternalTransfer # when from/to are both NOT equal to dao.
+  """
+  When a DAO is involved in a transfer and the source and destination addresses are both NOT equal to the DAO.
+  The DAO must have been approved for the token, and it's moving the token as an intermediary.
```

Comment:
```suggestion
  When a DAO facilitates a transfer and neither the source nor the destination addresses match the DAO's, it is essential that the DAO has approval to execute the token transfer, acting as an intermediary in the process.
```

Reply:
I like it, I will change it like this though:
  [...] it is essential that the DAO has token approval to execute the transfer [...]
  or [...] the DAO must have token approval to execute the transfer [...]